recent build failure for 22e7d1e

fails to build on Fedora 41 x86_64 with  gcc (GCC) 14.2.1 20240912 (Red Hat 14.2.1-3):

...
[628/675] Building C object code/client/cgame/CMakeFiles/cgame.dir/cg_marks.c.o
FAILED: code/client/cgame/CMakeFiles/cgame.dir/cg_marks.c.o 
/usr/bin/gcc -DCGAME_DLL -DNDEBUG -Dcgame_EXPORTS -I/tmp/openmohaa/code/qcommon/../qcommon -I/tmp/openmohaa/code/qcommon/../script -I/tmp/openmohaa/.cmake/code/qcommon/generated -fPIC -Wno-comment -Werror=return-type -fno-gnu-unique -MD -MT code/client/cgame/CMakeFiles/cgame.dir/cg_marks.c.o -MF code/client/cgame/CMakeFiles/cgame.dir/cg_marks.c.o.d -o code/client/cgame/CMakeFiles/cgame.dir/cg_marks.c.o -c /tmp/openmohaa/code/cgame/cg_marks.c
/tmp/openmohaa/code/cgame/cg_marks.c: In function ‘CG_AssembleFinalMarks’:
/tmp/openmohaa/code/cgame/cg_marks.c:802:52: error: passing argument 2 of ‘AngleVectorsLeft’ from incompatible pointer type [-Wincompatible-pointer-types]
  802 |                 AngleVectorsLeft(cent->lerpAngles, &axis[0], &axis[1], &axis[2]);
      |                                                    ^~~~~~~~
      |                                                    |
      |                                                    vec_t (*)[3] {aka float (*)[3]}
In file included from /tmp/openmohaa/code/cgame/cg_local.h:31,
                 from /tmp/openmohaa/code/cgame/cg_marks.c:26:
/tmp/openmohaa/code/qcommon/../qcommon/q_shared.h:895:52: note: expected ‘vec_t *’ {aka ‘float *’} but argument is of type ‘vec_t (*)[3]’ {aka ‘float (*)[3]’}
  895 | void AngleVectorsLeft( const vec3_t angles, vec3_t forward, vec3_t left, vec3_t up );
      |                                             ~~~~~~~^~~~~~~
/tmp/openmohaa/code/cgame/cg_marks.c:802:62: error: passing argument 3 of ‘AngleVectorsLeft’ from incompatible pointer type [-Wincompatible-pointer-types]
  802 |                 AngleVectorsLeft(cent->lerpAngles, &axis[0], &axis[1], &axis[2]);
      |                                                              ^~~~~~~~
      |                                                              |
      |                                                              vec_t (*)[3] {aka float (*)[3]}
/tmp/openmohaa/code/qcommon/../qcommon/q_shared.h:895:68: note: expected ‘vec_t *’ {aka ‘float *’} but argument is of type ‘vec_t (*)[3]’ {aka ‘float (*)[3]’}
  895 | void AngleVectorsLeft( const vec3_t angles, vec3_t forward, vec3_t left, vec3_t up );
      |                                                             ~~~~~~~^~~~
/tmp/openmohaa/code/cgame/cg_marks.c:802:72: error: passing argument 4 of ‘AngleVectorsLeft’ from incompatible pointer type [-Wincompatible-pointer-types]
  802 |                 AngleVectorsLeft(cent->lerpAngles, &axis[0], &axis[1], &axis[2]);
      |                                                                        ^~~~~~~~
      |                                                                        |
      |                                                                        vec_t (*)[3] {aka float (*)[3]}
/tmp/openmohaa/code/qcommon/../qcommon/q_shared.h:895:81: note: expected ‘vec_t *’ {aka ‘float *’} but argument is of type ‘vec_t (*)[3]’ {aka ‘float (*)[3]’}
  895 | void AngleVectorsLeft( const vec3_t angles, vec3_t forward, vec3_t left, vec3_t up );
      |                                                                          ~~~~~~~^~
[631/675] Building CXX object code/client/cgame/CMakeFiles/cgame.dir/cg_commands.cpp.o
ninja: build stopped: subcommand failed.


[openmohaa_22e7d1e_build_failure_gcc_14.2.1_fedora41_x86_64_20241029.txt](https://github.com/user-attachments/files/17563688/openmohaa_22e7d1e_build_failure_gcc_14.2.1_fedora41_x86_64_20241029.txt)

--- a/code/cgame/cg_marks.c
+++ b/code/cgame/cg_marks.c
@@ -799,7 +799,7 @@ void CG_AssembleFinalMarks(
 
                 // Fixed in OPM
                 //  Make the position completely local to the entity
-                AngleVectorsLeft(cent->lerpAngles, &axis[0], &axis[1], &axis[2]);
+                AngleVectorsLeft(cent->lerpAngles, axis[0], axis[1], axis[2]);
                 MatrixTransformVectorRight(axis, pos, pMark->pos);
             } else {
                 VectorCopy(pos, pMark->pos);

